terraform/main.tf

```diff
@@ -0,0 +1,89 @@
+# Copyright 2025 Canonical Ltd.
+# See LICENSE file for licensing details.
+
+# Juju model
+
+resource "juju_model" "dns" {
```

Comment:
If there is no need to change anything in the model, a data can be used instead of importing the resource.
Example:
https://github.com/canonical/wazuh-server-operator/blob/main/terraform/product/main.tf